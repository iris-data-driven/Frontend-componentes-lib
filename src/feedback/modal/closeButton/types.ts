export type CloseButtonType = {
  onClick?: () => void;
};
