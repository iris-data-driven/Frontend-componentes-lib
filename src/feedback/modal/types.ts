import { Styles } from 'react-modal';

type Size = 'medium' | 'large';

export type ModalProps = {
  isOpen: boolean;
  onCloseModal: () => void;
  onAfterClose?: () => void;
  onAfterOpen?: () => void;
  size?: Size;
  contentLabel?: string;
  notClosable?: boolean;
};

export type ModalStyle = (size: Size) => Styles;
