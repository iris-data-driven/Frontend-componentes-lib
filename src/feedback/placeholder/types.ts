type placeholderTypes = 'heading' | 'text' | 'circle' | 'custom';

export type PlaceholderStyleProps = {
  type?: placeholderTypes;
};

export type PlaceholderProps = PlaceholderStyleProps & {
  className?: string;
};
