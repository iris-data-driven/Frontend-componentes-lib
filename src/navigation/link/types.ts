export type LinkProps = {
  to: string;
  type?: 'default' | 'small';
};
