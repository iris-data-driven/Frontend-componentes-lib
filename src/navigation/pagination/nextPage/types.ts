export type NextPageProps = {
  currentPage: number;
  changePage: (page: number) => void;
  totalPages: number;
};
