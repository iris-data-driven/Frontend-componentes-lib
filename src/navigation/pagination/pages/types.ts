import { MouseEvent as ReactMouseEvent } from 'react';

export type PagesProps = {
  currentPage: number;
  totalPages: number;
  changePage: (page: number) => void;
};

export type PageButtonProps = {
  value: number;
  onClick: (event: ReactMouseEvent<HTMLButtonElement, MouseEvent>) => void;
};
