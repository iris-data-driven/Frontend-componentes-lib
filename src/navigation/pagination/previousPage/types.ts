export type PreviousPageProps = {
  currentPage: number;
  changePage: (page: number) => void;
};

export type WrapperType = {
  isClickable: boolean;
};
