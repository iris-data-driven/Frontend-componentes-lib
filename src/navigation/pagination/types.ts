export type PaginationProps = {
  totalPages: number;
  currentPage?: number;
  onChangePage: (page: number) => void;
};

export type WrapperButtonType = {
  isClickable: boolean;
};
