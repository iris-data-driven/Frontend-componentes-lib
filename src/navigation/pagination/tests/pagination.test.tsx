import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';

import Pagination from '../';

import { PaginationProps } from '../types';

describe('Pagination', () => {
  let props: PaginationProps;

  beforeEach(() => {
    props = {
      currentPage: 1,
      totalPages: 20,
      onChangePage: page => (props.currentPage = page),
    };
  });

  it('Should match the snapshot', () => {
    render(<Pagination {...props} />);

    const component = screen.getByRole('pagination');

    expect(component).toMatchSnapshot();
  });

  it('Should render nextPage and previousPage buttons', () => {
    render(<Pagination {...props} />);

    expect(screen.getByRole('next-page-button')).toBeInTheDocument();
  });

  it('Should return new page number when change page', () => {
    render(<Pagination {...props} />);

    fireEvent.click(screen.getByText('2'));

    expect(props.currentPage).toEqual(2);
  });

  it('Should render the initial stage when the page is less than 5', () => {
    render(<Pagination {...props} />);
    const components = screen.getAllByRole('paragraph');

    expect(components[0].innerHTML).toEqual('Anterior');
    expect(components[1].innerHTML).toEqual('1');
    expect(components[2].innerHTML).toEqual('2');
    expect(components[3].innerHTML).toEqual('3');
    expect(components[4].innerHTML).toEqual('4');
    expect(components[5].innerHTML).toEqual('5');
    expect(components[6].innerHTML).toEqual('...');
    expect(components[7].innerHTML).toEqual('20');
    expect(components[8].innerHTML).toEqual('Próxima');
  });

  it('Should render the medium stage when the page is more than 5', async () => {
    let newProps = props;

    newProps.currentPage = 6;

    render(<Pagination {...newProps} />);

    const components = screen.getAllByRole('paragraph');

    expect(components[0].innerHTML).toEqual('Anterior');
    expect(components[1].innerHTML).toEqual('1');
    expect(components[2].innerHTML).toEqual('...');
    expect(components[3].innerHTML).toEqual('4');
    expect(components[4].innerHTML).toEqual('5');
    expect(components[5].innerHTML).toEqual('6');
    expect(components[6].innerHTML).toEqual('...');
    expect(components[7].innerHTML).toEqual('20');
    expect(components[8].innerHTML).toEqual('Próxima');
  });

  it('Should render the final stage when the page is 5 less than the final page', async () => {
    let newProps = props;

    newProps.currentPage = 16;

    render(<Pagination {...newProps} />);

    const components = screen.getAllByRole('paragraph');

    expect(components[0].innerHTML).toEqual('Anterior');
    expect(components[1].innerHTML).toEqual('1');
    expect(components[2].innerHTML).toEqual('...');
    expect(components[3].innerHTML).toEqual('16');
    expect(components[4].innerHTML).toEqual('17');
    expect(components[5].innerHTML).toEqual('18');
    expect(components[6].innerHTML).toEqual('19');
    expect(components[7].innerHTML).toEqual('20');
    expect(components[8].innerHTML).toEqual('Próxima');
  });
});
