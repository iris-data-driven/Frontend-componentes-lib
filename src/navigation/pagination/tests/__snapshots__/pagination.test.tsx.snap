// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Should match the snapshot 1`] = `
<div
  class="sc-AxirZ evipuV"
  role="pagination"
>
  <div
    class="sc-AxiKw iTpMkZ"
  >
    <svg
      class="sc-AxhCb ewgWZW"
      enable-background="new 0 0 20.5 20.5"
      version="1.1"
      viewBox="0 0 20.5 20.5"
      x="0"
      xmlns="http://www.w3.org/2000/svg"
      y="0"
    >
      <path
        d="M9.4 10.3L13 6.7c.2-.2.2-.4.2-.6 0-.2-.1-.4-.2-.6-.3-.3-.9-.3-1.2 0L7.6 9.7c-.3.3-.3.9 0 1.2l4.2 4.2c.2.2.4.2.6.2.2 0 .4-.1.6-.3.3-.3.3-.8 0-1.2l-3.6-3.5z"
      />
    </svg>
    <p
      class="sc-AxjAm kFUFEV"
      role="paragraph"
      type="p4"
    >
      Anterior
    </p>
  </div>
  <div
    class="sc-AxhUy gWaiz"
  >
    <button
      class="sc-AxgMl lfEqdP"
      value="1"
    >
      <p
        class="sc-AxjAm bKicft"
        role="paragraph"
        type="p2"
      >
        1
      </p>
    </button>
    <button
      class="sc-AxgMl gCZbzB"
      value="2"
    >
      <p
        class="sc-AxjAm bKicft"
        role="paragraph"
        type="p2"
      >
        2
      </p>
    </button>
    <button
      class="sc-AxgMl gCZbzB"
      value="3"
    >
      <p
        class="sc-AxjAm bKicft"
        role="paragraph"
        type="p2"
      >
        3
      </p>
    </button>
    <button
      class="sc-AxgMl gCZbzB"
      value="4"
    >
      <p
        class="sc-AxjAm bKicft"
        role="paragraph"
        type="p2"
      >
        4
      </p>
    </button>
    <button
      class="sc-AxgMl gCZbzB"
      value="5"
    >
      <p
        class="sc-AxjAm bKicft"
        role="paragraph"
        type="p2"
      >
        5
      </p>
    </button>
    <p
      class="sc-AxjAm bKicft sc-AxheI bSgSTE"
      role="paragraph"
      type="p2"
    >
      ...
    </p>
    <button
      class="sc-AxgMl gCZbzB"
      value="20"
    >
      <p
        class="sc-AxjAm bKicft"
        role="paragraph"
        type="p2"
      >
        20
      </p>
    </button>
  </div>
  <div
    class="sc-AxiKw fQuMSW"
    role="next-page-button"
  >
    <p
      class="sc-AxjAm kFUFEV"
      role="paragraph"
      type="p4"
    >
      Pr√≥xima
    </p>
    <svg
      class="sc-Axmtr kyZvrQ"
      enable-background="new 0 0 20.5 20.5"
      version="1.1"
      viewBox="0 0 20.5 20.5"
      x="0"
      xmlns="http://www.w3.org/2000/svg"
      y="0"
    >
      <path
        d="M9.4 10.3L13 6.7c.2-.2.2-.4.2-.6 0-.2-.1-.4-.2-.6-.3-.3-.9-.3-1.2 0L7.6 9.7c-.3.3-.3.9 0 1.2l4.2 4.2c.2.2.4.2.6.2.2 0 .4-.1.6-.3.3-.3.3-.8 0-1.2l-3.6-3.5z"
      />
    </svg>
  </div>
</div>
`;
